### DISCUSSION TITLE
G-helper update drivers questions

### DISCUSSION THREAD

**Discussion by Chrystef** [seerge/g-helper#1350] [Q&A]
Greetings fellow g-helper users and of course my congrats to Mr. Seerge for his fantastic tool.

I am an 90% happy owner of G713PI with 7845hx & 4070. Still have some problems i am working (ex. screen is flashing sometimes..)

Below some questions about g-helper updates:
1) how do you suggest to update my drivers for the grey ones? should i re-enable asus services and go to live-update session and check?
2) for the green ones.. are asus verified? or it is considered as manual installation? is it safe to install them or should i try my asus live updates?
3) i have the 0.123 version and i had the option to update the bios to 327. I have done so, but g-helper updates windows still have it as 324..

![image](https://github.com/seerge/g-helper/assets/145780694/b150cced-5db7-4215-9d46-9359f7ad3b56)

some other questions:
I noticed that power limits does not apply if i don't hit the apply custom fans as well. true? or am i missing something..

Thank you for your time,
Best Regards,
Stefanos

**Comment by seerge** [comment]
@Chrystef hello

1. App gets list of drivers **AND** list of devices related to this drivers from Asus API (website).  For some drivers (and mainly for all software components like AMD Raid Driver, etc) Asus doesn't provide list of devices at all. So there is no way to check what version is currently installed and if it's installed at all. Those are marked with **grey**. MyASUS uses same approach. So it also won't be able to compare / check if there is a newer version available or not :) 
https://rog.asus.com/support/webapi/product/GetPDDrivers?website=global&model=G713PI&osid=52 

3. Green ones is where app has managed to check local version, and it's same or newer than one on Asus website.

4. On Asus website (and therefore in API) the latest one for your model is 324. Sometimes asus would push newer BIOS in portions, and then a couple of days (or weeks) later release to everyone :) 

<img width="1419" alt="Screenshot 2023-09-22 at 16 13 07" src="https://github.com/seerge/g-helper/assets/5920850/f7fda55e-9588-4ce2-9882-0a2c65d3ce35">

**Reply by Chrystef** [reply]
I love you Seerge! thank you for your immediate answer!

**Reply by MC-dusk** [reply]
So when the version is red, that means local version is older than the version in the asus online list (or call it api) ?
And green means local is newer or the same.
But grey means can't detect local version and the number shows online newest version in the list of api?
I wonder if I take it right.

**Reply by seerge** [reply]
Correct. Red is when update is available.

**Reply by YESNLars** [reply]
@seerge 
So if updates are red, do they automatically update?

**Comment by seerge** [comment]
@Chrystef as for power limits applying only together with fans - it is how some models work (apparently yours is one of them). As in Armoury you can apply both together or none. 

I can add your model to a list of devices where g-helper would silently apply a fan curve every time power limits are applied.

**Reply by Chrystef** [reply]
thank you!

**Reply by seerge** [reply]
@Chrystef this build should do the thing
[GHelper.zip](https://github.com/seerge/g-helper/files/12706266/GHelper.zip)

**Comment by Vikman83** [comment]
> 3\. i have the 0.123 version and i had the option to update the bios to 327. I have done so, but g-helper updates windows still have it as 324..

ASUS just removed 327 from the update site. Don't worry. I know for sure it was available on the site and now it’s gone

**Comment by LoveYourEnemy** [comment]
It is a bit relevant: Whenever I update driver(s) through Ghelper, all the previously closed Asus services would come back and running again. Every time after the driver update, I have to go to Extra to stop the services. Is this a bug or it is how it is?

**Reply by seerge** [reply]
@LoveYourEnemy it's not when you update "drivers", but when you update "Asus System Control Interface" that includes all those services and starts them again (obviously). 

You can stop them again from `Extra`.

**Reply by LoveYourEnemy** [reply]
Thank you for clarifying! It's good to know that it was this interface thing screwed up the maximum charge setting every time it updates. I will have to make sure to Login to windows immediately after the interface update from now on, otherwise laptop charges again to 100% when I afk after restart (which is horrible for the battery).
